src/Raven.Server/Documents/Indexes/Static/JavaScriptIndex.cs

```diff
@@ -218,8 +222,8 @@ private HashSet<CollectionName> ExecuteCodeAndCollectReferencedCollections(strin
 
             if (definition.Reduce != null)
             {
-                _engine.Execute(definition.Reduce);
-        }
+                _engine.ExecuteWithReset(definition.Reduce);
```

Comment:
There is no limit to the number of steps it takes to _define_ maps / reduce

Reply:
`InitializeEngine` is wrapped with `DisableMaxStatements`